spec/lib/que/worker_spec.rb

```diff
@@ -19,34 +19,84 @@
-               ))
+      context "logging" do
+        it "logs the work without custom log context" do
+          expect(Que.logger).to receive(:info).
+            with(hash_including(
+                   event: "que_job.job_begin",
+                   handler: "FakeJob",
+                   job_class: "FakeJob",
+                   job_error_count: 0,
+                   msg: "Job acquired, beginning work",
+                   priority: 100,
+                   queue: "default",
+                   que_job_id: job.attrs["job_id"],
+                   latency: an_instance_of(Float),
+                 ))
+
+          expect(Que.logger).to receive(:info).
+            with(hash_including(
+                   duration: kind_of(Float),
+                   event: "que_job.job_worked",
+                   handler: "FakeJob",
+                   job_class: "FakeJob",
+                   job_error_count: 0,
+                   msg: "Successfully worked job",
+                   priority: 100,
+                   queue: "default",
+                   que_job_id: job.attrs["job_id"],
+                 ))
+          subject
+        end
 
-        subject
+        context "with custom log context" do
+          let!(:job) do
+            class FakeJobWithCustomLogs < FakeJob
+              custom_log_context ->(attrs) {
+                {
+                  custom_log_1: attrs[:args][0],
+                  custom_log_2: "test-log",
+                }
+              }
+
+              @log = []
+            end
+            FakeJobWithCustomLogs.enqueue(1)
+          end
+
+          it "logs the work with custom log context" do
+            expect(Que.logger).to receive(:info).
+              with(hash_including(
+                     event: "que_job.job_begin",
+                     handler: "FakeJobWithCustomLogs",
+                     job_class: "FakeJobWithCustomLogs",
+                     job_error_count: 0,
+                     msg: "Job acquired, beginning work",
+                     priority: 100,
+                     queue: "default",
+                     que_job_id: job.attrs["job_id"],
+                     latency: an_instance_of(Float),
+                     custom_log_1: 1,
+                     custom_log_2: "test-log",
+                   ))
+
+            expect(Que.logger).to receive(:info).
+              with(hash_including(
+                     duration: kind_of(Float),
+                     event: "que_job.job_worked",
+                     handler: "FakeJobWithCustomLogs",
+                     job_class: "FakeJobWithCustomLogs",
+                     job_error_count: 0,
+                     msg: "Successfully worked job",
+                     priority: 100,
+                     queue: "default",
+                     que_job_id: job.attrs["job_id"],
+                     custom_log_1: 1,
+                     custom_log_2: "test-log",
+                   ))
+            subject
```

Comment:
Maybe we could name the subject to something like `work_job` to make the intent clearer here?

Reply:
I found this a bit weird too but it's using rspec's `subject` keyword so it can't just be renamed and I don't want to take things apart too much

Reply:
You can name a subject like so:

```rb
subject(:work_job) { described_class.new.work }
```

Which would allow you to refer to the subject either using `subject` or `work_job`.

Alternatively, you could just call `described_class.new.work` in your spec instead of `subject`. (Thinking about it, I think this is a better approach. Naming the subject `work_job` is actually a bit misleading because the job would only be worked the first time `work_job` would be called.)